packages/ember-glimmer/lib/views/outlet.js

```diff
@@ -25,6 +25,39 @@ class OutletStateReference {
   }
 }
 
+// So this is a relic of the past that SHOULD go away
+// in 3.0. Preferably it is deprecated in the release that
+// follows the Glimmer release.
+export class OrphanedOutletState {
+  constructor(name, inner, orphan) {
+    this.tag = inner.tag;
+    this.inner = inner;
+    this.orphan = orphan;
+    this.orphanState = orphan.get(name).value();
+    this.name = name;
+  }
+
+  value() {
+    let currentOutletState = this.inner.value();
+    let orphanState = this.orphan.get(this.name).value();
+
+    if (!orphanState.wasUsed) {
+      currentOutletState.outlets[this.name] = orphanState;
+      orphanState.wasUsed = true;
```

Comment:
We don't want to do these kind of mutations on a reference – can we move this mutation into the caller? (the code in Manager#create)